File: fixtures/objc/tink_aesgcm/src/TinkExample.m
#import <Foundation/Foundation.h>
#import <Tink/Tink.h>
#import "objc/TINKConfig.h"
#import "objc/TINKKeysetHandle.h"
#import "objc/TINKAeadFactory.h"
#import "objc/TINKAeadKeyTemplate.h"

int main() {
    @autoreleasepool {
        NSError *error = nil;
        
        // Register Tink configuration
        if (![TINKConfig registerConfig:&error]) {
            NSLog(@"Failed to register config: %@", error);
            return 1;
        }
        
        // Generate new AES-GCM keyset
        TINKAeadKeyTemplate *template = [[TINKAeadKeyTemplate alloc] 
            initWithKeyTemplate:TINKAes256Gcm error:&error];
        if (!template) {
            NSLog(@"Failed to create template: %@", error);
            return 1;
        }
        
        TINKKeysetHandle *keysetHandle = [[TINKKeysetHandle alloc] 
            initWithKeyTemplate:template error:&error];
        if (!keysetHandle) {
            NSLog(@"Failed to generate keyset: %@", error);
            return 1;
        }
        
        // Get AEAD primitive
        id<TINKAead> aead = [TINKAeadFactory primitiveWithKeysetHandle:keysetHandle 
                                                                  error:&error];
        if (!aead) {
            NSLog(@"Failed to get primitive: %@", error);
            return 1;
        }
        
        // Encrypt
        NSData *plaintext = [@"Hello Tink ObjC" dataUsingEncoding:NSUTF8StringEncoding];
        NSData *associatedData = [@"metadata" dataUsingEncoding:NSUTF8StringEncoding];
        NSData *ciphertext = [aead encrypt:plaintext 
                         withAdditionalData:associatedData 
                                      error:&error];
        if (!ciphertext) {
            NSLog(@"Encryption failed: %@", error);
            return 1;
        }
        
        // Decrypt
        NSData *decrypted = [aead decrypt:ciphertext 
                        withAdditionalData:associatedData 
                                     error:&error];
        if (!decrypted) {
            NSLog(@"Decryption failed: %@", error);
            return 1;
        }
        
        NSString *result = [[NSString alloc] initWithData:decrypted 
                                                  encoding:NSUTF8StringEncoding];
        NSLog(@"Decrypted: %@", result);
    }
    return 0;
}

File: fixtures/python/tink_aesgcm/src/tink_example.py
import tink
from tink import aead
from tink import cleartext_keyset_handle

def main():
    # Register AEAD key types
    aead.register()
    
    # Generate a new AES-GCM key
    key_template = aead.aead_key_templates.AES256_GCM
    keyset_handle = tink.new_keyset_handle(key_template)
    
    # Get the AEAD primitive
    aead_primitive = keyset_handle.primitive(aead.Aead)
    
    # Encrypt data
    plaintext = b'Hello Tink Python'
    associated_data = b'metadata'
    ciphertext = aead_primitive.encrypt(plaintext, associated_data)
    
    # Decrypt data
    decrypted = aead_primitive.decrypt(ciphertext, associated_data)
    print(f"Decrypted: {decrypted.decode()}")
    
    # Also demonstrate AES-CTR-HMAC
    ctr_template = aead.aead_key_templates.AES256_CTR_HMAC_SHA256
    ctr_handle = tink.new_keyset_handle(ctr_template)
    ctr_aead = ctr_handle.primitive(aead.Aead)
    
    # And ChaCha20-Poly1305
    chacha_template = aead.aead_key_templates.CHACHA20_POLY1305
    chacha_handle = tink.new_keyset_handle(chacha_template)
    chacha_aead = chacha_handle.primitive(aead.Aead)

if __name__ == '__main__':
    main()
